plugins/modules/ome_application_network_proxy.py

```diff
@@ -173,56 +252,102 @@ def validate_check_mode_for_network_proxy(payload_diff, module):
     remove_unwanted_keys(remove_keys, backup_params)
-    payload = dict([(proxy_payload_map[key], val) for key, val in backup_params.items() if val is not None])
+    payload = dict([(proxy_payload_map[key], val) for key, val in backup_params.items() if val
+                    is not None])
+    if backup_params.get("proxy_exclusion_list") or backup_params.get("proxy_exclusion_list") == []:
+        temp_proxy_exclusion_list = backup_params.get("proxy_exclusion_list")
+        converted_proxy_exclusion_list = ",".join(temp_proxy_exclusion_list)
+        payload["ProxyExclusionList"] = converted_proxy_exclusion_list
     return payload
 
 
 def get_updated_payload(rest_obj, module, payload):
+    """
+    Generates a payload for the network proxy configuration.
+
+    Args:
+        rest_obj (object): The REST object.
+        module (object): The Ansible module object.
+        payload (dict): The payload for the network proxy configuration.
+
+    Returns:
+        dict: The updated payload for the network proxy configuration.
+    """
     current_setting = {}
     if not any(payload):
-        module.fail_json(msg="Unable to configure the proxy because proxy configuration settings are not provided.")
+        module.fail_json(msg="Unable to configure the proxy because proxy configuration settings"
+                         " are not provided.")
     else:
         params = module.params
-        remove_keys = ["@odata.context", "@odata.type", "@odata.id", "Password"]
+        if params.get("update_password"):
+            remove_keys = ["@odata.context", "@odata.type", "@odata.id"]
+        else:
+            remove_keys = ["@odata.context", "@odata.type", "@odata.id", "Password"]
         enable_authentication = params.get("enable_authentication")
         if enable_authentication is False:
-            """when enable auth is disabled, ignore proxy username and password """
             remove_keys.append("Username")
             payload.pop('Username', None)
             payload.pop('Password', None)
+        else:
+            if params.get("update_password"):
+                payload['Username'] = params.get("proxy_username")
+                payload['Password'] = params.get("proxy_password")
         resp = rest_obj.invoke_request("GET", PROXY_CONFIG)
         current_setting = resp.json_data
         remove_unwanted_keys(remove_keys, current_setting)
-        diff = any(key in current_setting and val != current_setting[key] for key, val in payload.items())
+        diff = any(key in current_setting and val != current_setting[key] for key, val in
+                   payload.items())
         validate_check_mode_for_network_proxy(diff, module)
         if not diff:
-            module.exit_json(msg="No changes made to proxy configuration as entered values are the same as current "
-                                 "configuration values.")
+            module.exit_json(msg="No changes made to proxy configuration as entered values are "
```

Comment:
Move the error messages to global variables

Reply:
Done